src/main/webapp/app/exercises/shared/rating/rating.service.ts

```diff
@@ -0,0 +1,38 @@
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { Observable } from 'rxjs/Observable';
+import { Rating } from 'app/entities/rating.model';
+import { SERVER_API_URL } from 'app/app.constants';
+
+@Injectable({
+    providedIn: 'root',
+})
+export class RatingService {
+    private ratingResourceUrl = SERVER_API_URL + 'api/results/';
+
+    constructor(private http: HttpClient) {}
+
+    /**
+     * Update the student rating for feedback on the server.
+     * @param rating - Rating for the result
+     */
+    setRating(rating: Rating): Observable<Rating> {
+        return this.http.post<Rating>(this.ratingResourceUrl + `${rating.id}/rating/${rating.rating}`, rating);
+    }
+
+    /**
+     * Get rating for "resultId" Result
+     * @param resultId - Id of Result who's rating is received
+     */
+    getRating(resultId: number): Observable<Rating> {
+        return this.http.get<Rating>(this.ratingResourceUrl + `${resultId}/rating`);
+    }
+
+    /**
+     * Update rating for "resultId" Result
+     * @param rating - Rating for the result
+     */
+    updateRating(rating: Rating): Observable<Rating> {
+        return this.http.put<Rating>(this.ratingResourceUrl + `${rating.id}/rating/${rating.rating}`, rating);
```

Comment:
All the urls seem inconsistent with the server endpoints. Does `rating.service.ts` still need to be adapted to the latest changes in `RatingResource`?

Reply:
The result id needs to be passed in. `${rating.id}` should be `${result.id}`.

Reply:
Although I guess it's technically correct since they are the same when `MapsId` is used.

Reply:
I think the only real issue is in `setRating` then. It fails since `rating.id` is still undefined at that point.